src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1175,6 +1179,13 @@ private String getBranchName(Branch branch)
         return name;
     }
 
+
+    private String getShortBranchName(Branch branch) {
+        String name = firstNonNull(branch.getName(), "");
+        Iterable<String> parts = Splitter.on("/").omitEmptyStrings().split(name);
+        return Iterables.getLast(parts);
+    }
+
```

Comment:
Is that really what you want?  I use a branch named develop/pre_8.3 for the work on a branch that will eventually be the 8.3 release of the product.  It is coming from a remote named origin.  I believe that your algorithm will return "pre_8.3" when my branch name is "develop/pre_8.3".